nixos/modules/services/monitoring/prometheus/exporters/mail.nix

```diff
@@ -137,6 +139,13 @@ in
 {
   port = 9225;
   extraOpts = {
+    environmentFile = mkOption {
```

Comment:
First of all, it's for simplicity because otherwise you'd have to mess around with rendering the options into a JSON file even more. Second, you can define multiple servers with passphrases and you'd have to dynamically generate replacement strings for several passphrases which is also kinda messy.

So basically, if you have a single secret, `*File` + `replace-secret` is fine, otherwise I prefer `envsubst` (if the config format from a package doesn't support env vars already).

Reply:
Thanks, that makes sense.